src/Digdir.Domain.Dialogporten.Application/Common/Extensions/ClaimsPrincipalExtensions.cs

```diff
@@ -120,22 +192,56 @@ public static bool TryGetAuthenticationLevel(this ClaimsPrincipal claimsPrincipa
             c.Type.StartsWith(AltinnClaimPrefix, StringComparison.Ordinal)
         ).OrderBy(c => c.Type);
 
-    private static bool TryGetOrgShortName(this ClaimsPrincipal claimsPrincipal, [NotNullWhen(true)] out string? orgShortName)
-        => claimsPrincipal.FindFirst(OrgClaim).TryGetOrgShortName(out orgShortName);
+    public static (UserIdType, string externalId) GetUserType(this ClaimsPrincipal claimsPrincipal)
+    {
+        if (claimsPrincipal.TryGetPid(out var externalId))
+        {
+            return (claimsPrincipal.HasScope(ServiceProviderScope)
+                ? UserIdType.ServiceOwnerOnBehalfOfPerson
+                : UserIdType.Person, externalId);
+        }
+
+        if (claimsPrincipal.TryGetLegacySystemUserId(out externalId))
+        {
+            return (UserIdType.LegacySystemUser, externalId);
+        }
 
-    private static bool TryGetOrgShortName(this Claim? orgClaim, [NotNullWhen(true)] out string? orgShortName)
+        // https://docs.altinn.studio/authentication/systemauthentication/
+        if (claimsPrincipal.TryGetSystemUserId(out externalId))
+        {
+            return (UserIdType.SystemUser, externalId);
+        }
+
+        if (claimsPrincipal.HasScope(ServiceProviderScope) &&
+            claimsPrincipal.TryGetServiceOwnerShortName(out externalId))
```

Comment:
This assumes the `urn:altinn:org`-claim is always present, which only happens when the Maskinporten-token has particular scopes and is issued by Altinn Token Exchange (this is not mandatory). We should therefore not rely on this claim at all, and perform all service owner name resolution in `IServiceOwnerNameRegistry`. Thus, the `TryGetServiceOwnerShortName` methods in `ClaimsPrincipalExtensions ` including all usage of `OrgClaim` should be removed. `externalId` for service owners should here be the organization number.

Reply:
Good point, I didn't even realize we had two different short name lookups 🙈 Removed! 

But we do keep the same claim use, `private const string AttributeIdOrg = "urn:altinn:org";` in `DecisionRequestHelper` for building the xacmls?